scripts/functions/diagnostic_application.sh

```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+#* This file is part of the MOOSE framework
+#* https://www.mooseframework.org
+#*
+#* All rights reserved, see COPYRIGHT for full restrictions
+#* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+#*
+#* Licensed under LGPL 2.1, please see LICENSE for details
+#* https://www.gnu.org/licenses/lgpl-2.1.html
+
+function enter_moose()
+{
+    # TODO: allow a --use-moose-dir argument
+    cd $CTMP_DIR/moose || exit 1
+}
+
+function clone_moose()
+{
+    # TODO: allow a --use-moose-dir argument
+    printf "Creating a clean clone of MOOSE repository\n"
+    git clone --depth 1 https://github.com/idaholab/moose ${CTMP_DIR}/moose -b master &> ${CTMP_DIR}/moose_clone_stdouterr.log
```

Comment:
this failed on my machine
```
###########################################################################
Creating a clean clone of MOOSE repository
WARNING: SSL issues detected, attempting again with SSL protections off

This may indicate the cause of other issues. PETSc contribs may fail to
download properly.

export GIT_SSL_NO_VERIFY=true

Creating a clean clone of MOOSE repository
fatal: cannot change to '/var/folders/0j/yjnmnqfn6m1_q77kp_3k44jh27t6pk/T/MOOSE.Le5fBjfOkA/moose': No such file or directory
fatal: cannot change to '/var/folders/0j/yjnmnqfn6m1_q77kp_3k44jh27t6pk/T/MOOSE.Le5fBjfOkA/moose': No such file or directory
fatal: cannot change to '/var/folders/0j/yjnmnqfn6m1_q77kp_3k44jh27t6pk/T/MOOSE.Le5fBjfOkA/moose': No such file or directory

```